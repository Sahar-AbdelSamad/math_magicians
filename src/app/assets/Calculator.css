.calculator-route {
  display: flex;
  justify-content: space-between;
}

.calculator {
  width: 280px;
  margin-right: 10%;
}

.result {
  background-color: #a46596;
  padding-right: 5px;
  color: white;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.buttons {
  display: grid;
  grid-template-columns: repeat(4, 70px);
  grid-template-rows: repeat(5, 40px);
  background-color: #dfdfdf;
}

.buttons button {
  border: 1px solid #c483b5;
}

.buttons button:nth-child(17) {
  grid-area: 5/1/5/3;
}

.buttons button:nth-child(4n),
.buttons button:last-child {
  background-color: #d100a4;
}
