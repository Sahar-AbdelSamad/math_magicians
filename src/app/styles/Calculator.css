body {
  font-size: 12px;
}

.result {
  background-color: #858293;
  padding-right: 5px;
  color: white;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.buttons {
  display: grid;
  grid-template-columns: repeat(4, 70px);
  grid-template-rows: repeat(5, 40px);
  background-color: #dfdfdf;
}

.buttons button {
  border: 1px solid #d0d0d0;
}

.zero {
  grid-area: 5/1/5/3;
}

.buttons button:nth-child(4n),
.buttons button:last-child {
  background-color: #ff8e00;
}
