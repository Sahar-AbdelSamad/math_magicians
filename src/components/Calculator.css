body {
  font-size: 12px;
}

.table {
  border-collapse: collapse;
}

td {
  border: 1px solid #d0d0d0;
  width: 60px;
  height: 40px;
  text-align: center;
}

.input {
  background-color: #858293;
  width: 252px;
  height: 40px;
  border: none;
  padding: 0;
  padding-right: 5px;
  text-align: end;
  color: white;
}

tr {
  background-color: #dfdfdf;
}

td:last-child:not(:first-of-type) {
  background-color: #ff8e00;
}
