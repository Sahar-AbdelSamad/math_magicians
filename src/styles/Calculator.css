body {
  font-size: 12px;
}

.result {
  background-color: #858293;
  padding-right: 5px;
  color: white;
  text-align: end;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

button {
  width: 70px;
  height: 40px;
  border: 1px solid #d0d0d0;
  background-color: #dfdfdf;
}

.zero {
  width: 140px;
}

div button:last-child {
  background-color: #ff8e00;
}
