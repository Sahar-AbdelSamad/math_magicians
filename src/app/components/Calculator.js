import react from 'react';
import Button from './Button';
import calculate from '../logic/calculate';
import '../styles/Calculator.css';

export default class Calculator extends react.Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

  onClick = (e) => {
    this.setState((state) => calculate(state, e.target.innerText));
  };

  render() {
    const btn = {
      value: ['AC', '+/-', '%', '+', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', ' ='],
    };
    return (
      <div className="calculator">
        <div className="result">
          {this.state.total}
          {this.state.operation}
          {this.state.next}
        </div>
        <Button btn={btn} click={this.onClick} />
      </div>
    );
  }
}
