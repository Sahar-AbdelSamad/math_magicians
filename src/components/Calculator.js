import '../styles/Calculator.css';

const react = require('react');

export default class Calculator extends react.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <table className="table">
        <tr>
          <td colSpan="4"><input className="input" value="0" /></td>
        </tr>
        <tr>
          <td>AC</td>
          <td>+/-</td>
          <td>%</td>
          <td>+</td>
        </tr>
        <tr>
          <td>7</td>
          <td>8</td>
          <td>9</td>
          <td>*</td>
        </tr>
        <tr>
          <td>4</td>
          <td>5</td>
          <td>6</td>
          <td>-</td>
        </tr>
        <tr>
          <td>1</td>
          <td>2</td>
          <td>3</td>
          <td>+</td>
        </tr>
        <tr>
          <td colSpan="2">0</td>
          <td>.</td>
          <td>=</td>
        </tr>
      </table>
    );
  }
}
