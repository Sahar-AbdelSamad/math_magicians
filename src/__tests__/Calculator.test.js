import renderer from 'react-test-renderer';
import Calculator from '../app/components/Calculator';

it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
