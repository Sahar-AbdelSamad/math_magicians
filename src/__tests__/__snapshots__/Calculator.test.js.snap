// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
       className="calculator-route"
     >
       <h2>
         Let's do some Math!
       </h2>
       <div
         className="calculator"
       >
         <div
           className="result"
         >
           0
         </div>
         <div
           className="buttons"
         >
           <button
             onClick={[Function]}
             type="button"
           >
             AC
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             /-
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             %
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             รท
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             7
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             8
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             9
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             x
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             4
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             5
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             6
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             -
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             1
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             2
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             3
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             0
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
             .
           </button>
           <button
             onClick={[Function]}
             type="button"
           >
              =
           </button>
         </div>
       </div>
     </div>
`;

exports[`renders correctly 1`] = `
<div
  className="calculator-route"
>
  <h2>
    Let's do some Math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="result"
    >
      0
    </div>
    <div
      className="buttons"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         =
      </button>
    </div>
  </div>
</div>
`;
